### PR TITLE
systemd service fix

### PR DESCRIPTION
**Pull request by arthurtiteica**
Added systemd tmpfile (to be installed in '/usr/lib/tmpfiles.d').
Cleaned up the systemd service file to no longer include commands
handled by systemd-tmpfiles.

### CODE REVIEW

File: other/bootstrap_daemon/tox-bootstrapd.service [hunk 1]
```diff
@@ -4,11 +4,8 @@ After=network.target
 
 [Service]
 Type=forking
-PermissionsStartOnly=true
-ExecStartPre=-/bin/mkdir /var/run/tox-bootstrapd -p
-ExecStartPre=/bin/chown tox-bootstrapd:tox-bootstrapd -R /var/run/tox-bootstrapd
 WorkingDirectory=/var/lib/tox-bootstrapd
-ExecStart=/usr/local/bin/tox-bootstrapd /etc/tox-bootstrapd.conf
+ExecStart=/usr/bin/tox-bootstrapd /etc/tox-bootstrapd.conf
```
Comment (nurupo) [inline other/bootstrap_daemon/tox-bootstrapd.service]: Don't replace `local` with `bin`. We assume that you build the code yourself and install it yourself, and self-built programs go into `local` by default.

Comment (arthurtiteica) [inline other/bootstrap_daemon/tox-bootstrapd.service]: How will I get this reverted? I just make another commit with the path fixed?

Anyway, the systemd files get copied manually. Maybe I better make some amendments in the readme at other/bootstrap_daemon/README.md

What do you think?

**Comment by nurupo** [comment]
For the same reason as in https://github.com/irungentoo/toxcore/pull/1332#discussion_r28987628, the tempfile should be put in [`/etc/tmpfiles.d`](http://www.freedesktop.org/software/systemd/man/tmpfiles.d.html)

>Files in /etc/tmpfiles.d are reserved for the local administrator, who may use this logic to override the configuration files installed by vendor packages.

**Comment by nurupo** [comment]
Also, document that tempfiles should be put in there in the README file.

File: other/bootstrap_daemon/tox-bootstrapd-tmpfile.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+d /var/run/tox-bootstrapd 0750 tox-bootstrapd tox-bootstrapd -
```
Comment (nurupo) [inline other/bootstrap_daemon/tox-bootstrapd-tmpfile.conf]: There is no need to restrict a directory storing a pid file to `tox-bootstrapd` user:group.

Comment (nurupo) [inline other/bootstrap_daemon/tox-bootstrapd-tmpfile.conf]: Actually, ignore the comment that I just deleted here. Keep it as it is for now.

Comment (nurupo) [inline other/bootstrap_daemon/tox-bootstrapd-tmpfile.conf]: I see [it's suggested](http://www.freedesktop.org/software/systemd/man/tmpfiles.d.html) to use `RuntimeDirectory=` (and `RuntimeDirectoryMode=`) instead of tempfile for PID files, sockets, etc.

>System daemons frequently require private runtime directories below /run to place communication sockets and similar in. For these, consider declaring them in their unit files using RuntimeDirectory= (see systemd.exec(5) for details), if this is feasible.

File: other/bootstrap_daemon/tox-bootstrapd.service (hunk 1 above)
Comment (nurupo) [inline other/bootstrap_daemon/tox-bootstrapd.service]: >How will I get this reverted? I just make another commit with the path fixed?

Just make a commit and rebase (squash) it.

>Anyway, the systemd files get copied manually. Maybe I better make some amendments in the readme at other/bootstrap_daemon/README.md

Which amendments? any software you build from the source is set to install in `/usr/local/bin`, it's not Tox-specific, it's universal, there is no need to mention that.
If it's not built from source by the user, i.e. the user gets tox-boostrapd from distribution's package repository, then it's package maintainer's job to make sure that the software plays by distribution's rules and also that it installs in `/usr/bin`.

**Comment by arthurtiteica** [comment]
Personally I'm not willing to debate /usr/ versus /usr/local/. Choosing one or the other is wrong and to solve this these files should be generated at build time with the prefix chosen by the user.

Anyway there are many more valid prefixes, such as ~/.local/bin or /opt/

The sole reason I made this pull request is to point out the strange ExecPre lines in the service file and to suggest that those commands belong to tmpfiles in systemd.

That being said anything more than that is above my skill set so... sorry.

**Comment by stqism** [comment]
@arthurtiteica Hardly a debate, what @nurupo was mentioning is defined and standardized in the [Linux standard base](http://www.linuxfoundation.org/collaborate/workgroups/lsb).

**Comment by nurupo** [comment]
@irungentoo this PR can be closed, my PR (#1338) incorporated things discussed in here.
